vaadin-button-flow-parent/vaadin-button-flow/src/main/java/com/vaadin/flow/component/button/Button.java

```diff
@@ -271,15 +271,17 @@ public boolean isIconAfterText() {
     }
 
     /**
-     * Simulates a click on this button on the server side. Calling this method
-     * executes all registered click listeners on the server side, but does not
-     * execute possible client side registered listeners.
+     * Simulates a click on this button on the server side if it is enabled.
+     * Calling this method executes all registered click listeners on the server
+     * side, but does not execute possible client side registered listeners.
      *
      * @see #clickInClient()
      */
     public void click() {
-        fireEvent(new ClickEvent<>(this, false, 0, 0, 0, 0, 0, 0, false, false,
-                false, false));
+        if (isEnabled()) {
+            fireEvent(new ClickEvent<>(this, false, 0, 0, 0, 0, 0, 0, false,
+                    false, false, false));
```

Comment:
**note:** It seems that you can still get click triggered on a disabled button by manually firing this click event.